### DISCUSSION TITLE
Corrupt data error with array containing \u0651\u0650

### DISCUSSION THREAD

**Discussion by antja0** [livewire/livewire#5386] [7. Bugs (archived)] [answered]
EDIT:
Works on chrome, not in safari 14.0.3

## How to reproduce

Write this in `mount()` or `render()` in any livewire component:

`$this->someArray = json_decode('{"foo":"\u0651\u0650"}', true);`

Also add the property
`public $someArray = [];`

This will result in an error:
```
Livewire encountered corrupt data when trying to hydrate the [foobar] component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.
```

## Additional information

I noticed this happens only with certain characters / combinations.

I noticed that in example above if you put space in between these two it works:

- "\u0651 \u0650" works.
- "\u0651\u0650" doesn't work.

Livewire version: 2.10.7
Php version: 8.1

I guess I could get around this error by storing this code notation (\u0651\u0650) in livewire component and rendering it as arabic characters in "frontend", but if livewire would be able to handle this it would make things a lot simpler.
Or, I would show more detailed error message like "These characters cause problems" - as this error was rather difficult to find.

Thank you for developing livewire.

**Comment by joshhanley** [comment, marked as answer]
@antja0 there were issues with special characters, but I believe that has been fixed in PR #4942. That fix hasn't been released yet, but could you install the master branch in your app and test it to see if it helps? Thanks!

**Reply by antja0** [reply]
Thank you for response @joshhanley.

I happed to notice that the PR you linked mentioned Safari, so I tried with Chrome and turns out it's safari problem only. I have pretty old version of safari too: ( 14.0.3 ). My apologies, should've tried this beforehand.

Unfortunately updating livewire to master didn't help.

Then I happened to notice this linked open PR https://github.com/livewire/livewire/pull/5379 from the same contributor @balsarchive.
After installing that branch, it works! 🎉

**Reply by joshhanley** [reply]
@antja0 ok, thanks for testing that PR and letting us know 🙂
